garden-backend-service/src/api/routes/users.py

```diff
@@ -0,0 +1,37 @@
+from typing import Any, Dict
+
+from fastapi import APIRouter, Depends, status
+from sqlalchemy.ext.asyncio import AsyncSession
+from src.api.dependencies.auth import authed_user
+from src.api.dependencies.database import get_db_session
+from src.api.schemas.user import UserMetadataResponse, UserUpdateRequest
+from src.models import User
+
+router = APIRouter(prefix="/users")
+
+
+@router.get("", status_code=status.HTTP_200_OK, response_model=UserMetadataResponse)
+async def get_user_info(
+    authed_user: User = Depends(authed_user),
+):
+    """Return the user profile info for the current authed user."""
+    return authed_user
+
+
+@router.patch("", status_code=status.HTTP_200_OK, response_model=UserMetadataResponse)
+async def update_user_info(
+    user_update: UserUpdateRequest,
+    db: AsyncSession = Depends(get_db_session),
+    authed_user: User = Depends(authed_user),
+):
+    """Update the user profile info for the current authed user."""
+    update_data: Dict[str, Any] = user_update.dict(exclude_unset=True)
```

Comment:
style nits for a python 3.11 / pydantic 2.x world 
```suggestion
    update_data: dict[str, Any] = user_update.model_dump(exclude_unset=True)
```